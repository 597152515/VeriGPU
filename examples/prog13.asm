# negative offsets for load and store

li x1, 111
sw x1, 200(x0)

li x1, 222
sw x1, 300(x0)

li x3, 500

lw x1, -300(x3)
outr x1

lw x1, -200(x3)
outr x1

li x1, 444
sw x1, -300(x3)

li x1, 555
sw x1, -200(x3)

lw x1, 200(x0)
outr x1

lw x1, 300(x0)
outr x1

halt
