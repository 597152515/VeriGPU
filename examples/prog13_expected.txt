out 0 0000006f 111
out 1 000000de 222
out 2 000001bc 444
out 3 0000022b 555